File: javascript/examples/usage.ts
/**
 * CiteKit JavaScript SDK — Example Usage.
 *
 * Before running:
 *   1. Set GEMINI_API_KEY environment variable
 *   2. Run: npx tsx examples/usage.ts path/to/sample.pdf
 */

import { CiteKitClient } from "../src/index.js";

async function main() {
    const pdfPath = process.argv[2];
    if (!pdfPath) {
        console.log("Usage: npx tsx examples/usage.ts <path-to-pdf>");
        process.exit(1);
    }

    console.log(`📄 Resource: ${pdfPath}\n`);

    const client = new CiteKitClient();

    // ── Step 1: Ingest ────────────────────────────────────────────────────

    console.log("🔍 Ingesting resource...");
    const resourceMap = await client.ingest(pdfPath, "document");

    console.log(`✅ Map generated: ${resourceMap.resource_id}`);
    console.log(`   Title: ${resourceMap.title}`);
    console.log(`   Nodes: ${resourceMap.nodes.length}\n`);

    // ── Step 2: Inspect ───────────────────────────────────────────────────

    console.log("📋 Resource Map:");
    console.log("-".repeat(60));
    for (const node of resourceMap.nodes) {
        console.log(`  [${node.type}] ${node.id}`);
        console.log(`    Title: ${node.title}`);
        console.log(`    Pages: ${JSON.stringify(node.location.pages ?? [])}`);
        if (node.summary) console.log(`    Summary: ${node.summary}`);
        console.log();
    }

    // ── Step 3: Resolve ───────────────────────────────────────────────────

    if (resourceMap.nodes.length > 0) {
        const firstNode = resourceMap.nodes[0];
        console.log(`📎 Resolving node: ${firstNode.id}`);

        const evidence = await client.resolve(resourceMap.resource_id, firstNode.id);
        console.log(`  ✅ Output: ${evidence.output_path}`);
        console.log(`  📍 Address: ${evidence.address}`);
        console.log(`  📦 Modality: ${evidence.modality}`);
    }

    // ── Step 4: List maps ─────────────────────────────────────────────────

    console.log(`\n💾 Stored maps: ${JSON.stringify(client.listMaps())}`);
}

main().catch(console.error);

File: python/examples/usage.py
"""Example usage of CiteKit Python SDK.

Before running:
    1. pip install -e .
    2. Set your Gemini API key: export GEMINI_API_KEY=your_key_here
    3. Place a sample PDF in the same directory or adjust the path below.

Usage:
    python examples/usage.py path/to/sample.pdf
"""

import asyncio
import json
import os
import sys

from citekit import CiteKitClient, GeminiMapper


async def main():
    # ── Configuration ────────────────────────────────────────────────────────

    api_key = os.environ.get("GEMINI_API_KEY")
    if not api_key:
        print("Error: Set GEMINI_API_KEY environment variable.")
        sys.exit(1)

    if len(sys.argv) < 2:
        print("Usage: python examples/usage.py <path-to-pdf>")
        sys.exit(1)

    pdf_path = sys.argv[1]
    print(f"📄 Resource: {pdf_path}\n")

    # ── Step 1: Initialize ───────────────────────────────────────────────────

    mapper = GeminiMapper(api_key=api_key, model="gemini-2.0-flash")
    client = CiteKitClient(mapper=mapper)

    # ── Step 2: Ingest ───────────────────────────────────────────────────────

    print("🔍 Ingesting resource...")
    resource_map = await client.ingest(pdf_path, "document")

    print(f"✅ Map generated: {resource_map.resource_id}")
    print(f"   Title: {resource_map.title}")
    print(f"   Nodes: {len(resource_map.nodes)}\n")

    # ── Step 3: Inspect the map ──────────────────────────────────────────────

    print("📋 Resource Map:")
    print("-" * 60)
    for node in resource_map.nodes:
        pages = node.location.pages or []
        print(f"  [{node.type}] {node.id}")
        print(f"    Title: {node.title}")
        print(f"    Pages: {pages}")
        if node.summary:
            print(f"    Summary: {node.summary}")
        print()

    # ── Step 4: Resolve the first node ───────────────────────────────────────

    if resource_map.nodes:
        first_node = resource_map.nodes[0]
        print(f"📎 Resolving node: {first_node.id}")

        evidence = client.resolve(resource_map.resource_id, first_node.id)

        print(f"  ✅ Output: {evidence.output_path}")
        print(f"  📍 Address: {evidence.address}")
        print(f"  📦 Modality: {evidence.modality}")

    # ── Step 5: Show that maps persist ───────────────────────────────────────

    print(f"\n💾 Stored maps: {client.list_maps()}")


if __name__ == "__main__":
    asyncio.run(main())
